support/cas-server-support-duo/src/main/java/org/apereo/cas/adaptors/duo/config/DuoProviderRegistrationConfiguration.java

```diff
@@ -0,0 +1,59 @@
+package org.apereo.cas.adaptors.duo.config;
+
+import org.apereo.cas.configuration.model.support.mfa.DuoSecurityMultifactorProperties;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.MutablePropertyValues;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.beans.factory.support.RootBeanDefinition;
+import org.springframework.boot.context.properties.bind.Bindable;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+/**
+ * Registers Duo Provider beans into the registry.
+ *
+ * @author Travis Schmidt
+ * @since 6.0
+ */
+@Configuration("duoProviderRegistrationConfiguration")
+@Slf4j
+public class DuoProviderRegistrationConfiguration implements BeanDefinitionRegistryPostProcessor, EnvironmentAware {
+
+    private Environment environment;
+
+    @Override
+    public void postProcessBeanDefinitionRegistry(final BeanDefinitionRegistry beanDefinitionRegistry) throws BeansException {
+        val proxy = (RootBeanDefinition) beanDefinitionRegistry.getBeanDefinition("duoMfaProviderFactoryBean");
```

Comment:
This stuff really is dangerous and ultimately not maintainable; Way too many assumptions about the factory bean, and how it's proxied and plenty of other assumptions about cas property prefixes, etc.

I can  try to help, if you could explain what exactly needs to happen here; (perhaps we should chat on zoom, etc); and the explanations should be in javadocs too :)

Reply:
Agreed, this was my biggest concern as well.  The intent is to create beans that are registered at run time due to configuration, and have them participate in @RefreshScope.  This was the only way I was able to find to make it happen.